old-ui/app/components/fiat-value.js

```diff
@@ -1,71 +1,98 @@
-const Component = require('react').Component
-const h = require('react-hyperscript')
-const inherits = require('util').inherits
-const formatBalance = require('../util').formatBalance
+import React, { Component } from 'react'
+import { formatBalance } from '../util'
+import PropTypes from 'prop-types'
 
-module.exports = FiatValue
+class FiatValue extends Component {
+  render = () => {
+    const props = this.props
+    let { conversionRate } = props
+    const { currentCurrency, network } = props
+    const isSokol = parseInt(network) === 77
+    const isDai = parseInt(network) === 100
+    if (isSokol) {
+      conversionRate = 0
+    } else if (isDai) {
+      conversionRate = 1
+    }
+    const renderedCurrency = currentCurrency || ''
 
-inherits(FiatValue, Component)
-function FiatValue () {
-  Component.call(this)
-}
+    const value = formatBalance(props.value, 6, undefined, props.network)
 
-FiatValue.prototype.render = function () {
-  const props = this.props
-  let { conversionRate } = props
-  const { currentCurrency, network } = props
-  const isSokol = parseInt(network) === 77
-  const isDai = parseInt(network) === 100
-  if (isSokol) {
-    conversionRate = 0
-  } else if (isDai) {
-    conversionRate = 1
-  }
-  const renderedCurrency = currentCurrency || ''
+    if (value === 'None') return value
+    var fiatDisplayNumber, fiatTooltipNumber
+    var splitBalance = value.split(' ')
 
-  const value = formatBalance(props.value, 6, undefined, props.network)
+    fiatTooltipNumber = Number(splitBalance[0]) * conversionRate
+    fiatDisplayNumber = fiatTooltipNumber.toFixed(this.countSignificantDecimals(fiatTooltipNumber, 2))
 
-  if (value === 'None') return value
-  var fiatDisplayNumber, fiatTooltipNumber
-  var splitBalance = value.split(' ')
+    const valueStyle = props.valueStyle ? props.valueStyle : {
+      width: '100%',
+      textAlign: 'right',
+      fontSize: '14px',
+      color: '#ffffff',
+    }
 
-  fiatTooltipNumber = Number(splitBalance[0]) * conversionRate
-  fiatDisplayNumber = fiatTooltipNumber.toFixed(2)
+    const dimStyle = props.dimStyle ? props.dimStyle : {
+      color: '#60db97',
+      marginLeft: '5px',
+      fontSize: '14px',
+    }
 
-  const valueStyle = props.valueStyle ? props.valueStyle : {
-    width: '100%',
-    textAlign: 'right',
-    fontSize: '14px',
-    color: '#ffffff',
+    return this.fiatDisplay(fiatDisplayNumber, valueStyle, dimStyle, renderedCurrency.toUpperCase())
   }
 
-  const dimStyle = props.dimStyle ? props.dimStyle : {
-    color: '#60db97',
-    marginLeft: '5px',
-    fontSize: '14px',
+  /**
+   * returns the length of truncated significant decimals for fiat value
+   * @param {float} val The float value to be truncated
+   * @param {number} len The length of significant decimals
+   * returns {number} The length of truncated significant decimals
+  **/
+  countSignificantDecimals = (val, len) => {
```

Comment:
Does it makes sense to extract `countSignificantDecimals` from the react component into some kind of `utils` file so it can be reused later in other components?

Reply:
@patitonar Yes, it makes sense, thank you. Moved to `old-ui/app/util.js`, where common functions for components are stored.